search2/python/search/save/listener.py

```diff
@@ -1,194 +1,40 @@
-import ray
-import time
-import math
-import multiprocessing
-from ray.util.multiprocessing import Pool
-import json
-import pyarrow
-import gzip
-from ray.cluster_utils import Cluster
-import time
-import pickle
-import json
-from collections.abc import Iterable
-from pystalk import BeanstalkClient, BeanstalkError
-from .handler import go
-from typing import ByteString
-from os import path
-import pprint
-from orjson import loads, dumps
-from ruamel.yaml import YAML
+"""TODO: Add description here"""
+
 import argparse
-import glob
-from opensearchpy.exceptions import NotFoundError
-import traceback
-import time
+from typing import Any
+
+from ruamel.yaml import YAML
+
+from search.save.handler import go
+from search.utils.beanstalkd import QueueConf, listen, Publisher, get_config_file_path
+
 required_keys = ("outputBasePath", "assembly", "queryBody", "fieldNames", "indexName")
 optional_keys = ("indexConfig", "pipeline")
 
-
 def _coerce_inputs(
-    job_details,
-    job_id,
-    publisher_host,
-    publisher_port,
-    progress_event,
-    event_queue,
-    config_path_base_dir,
+    job_details: dict,
 ):
     job_specific_args = {}
 
     for key in required_keys:
         if key not in job_details:
-            raise Exception(f"Missing required key: {key} in job message")
+            raise ValueError(f"Missing required key: {key} in job message")
```

Comment:
This is great, and we could go even further in the spirit of "parse, don't validate" by creating a config class that can only be constructed by passing this validation check-- once it's constructed, we know forevermore that it's correct.

Reply:
What is the business value in doing this for a single use with hardcoded required, optional keys? That config class would almost certainly just replicate this, and we would use it once.

Reply:
Yeah,  it's possible its lifetime isn't long enough to justify the lift here.